### ISSUE TITLE
Bug aria-labelledby property is not currently supported to be added to progressbar component

### ISSUE THREAD

**Issue by jsdev** [react-bootstrap/react-bootstrap#6738]
### Prerequisites

- [X] I am using the [correct version](https://github.com/react-bootstrap/react-bootstrap#bootstrap-compatibility) of React-Bootstrap for my version of Bootstrap
- [X] I have [searched](https://github.com/react-bootstrap/react-bootstrap/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/react-bootstrap/react-bootstrap/blob/master/CONTRIBUTING.md)

### Describe the bug

progress bar fail to be labeled by assistive technology

I have included a link: of what should be possible to render:

https://codepen.io/jsdev/full/YzBoyqL

### Expected behavior

https://codepen.io/jsdev/full/YzBoyqL

you can see progressbar can be labelled by a heading above, doesn't have to be a heading this is an example of what should be possible and is not.

currently label only inserted as a text node, that is not translated to assistive technology as an accessible-name for the progress bar

### To Reproduce

https://react-bootstrap.netlify.app/docs/components/progress

it's pervasive it's never been supported.

I have a PR and will link to this issue

### Reproducible Example

https://react-bootstrap.netlify.app/docs/components/progress

### Screenshots

<img width="1341" alt="Screenshot 2023-12-14 at 1 50 44 PM" src="https://github.com/react-bootstrap/react-bootstrap/assets/1638860/2888e2f2-d0cf-44cb-a351-e5ca76061e30">

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of React-Bootstrap are you using?

5

### What version of Bootstrap are you using?

5.1.3

**Comment by jsdev** [comment]
https://codesandbox.io/p/sandbox/focused-cdn-c585fv?file=%2Fsrc%2FDemo.tsx%3A39%2C8

here is an example of competing react component library that allows aria-label and aria-labelledby correctly.

this can be solved multiple ways can lock down the props and or just pass props down to child progressbar component instead of progressbar wrapper component

<img width="649" alt="Screenshot 2023-12-14 at 3 17 31 PM" src="https://github.com/react-bootstrap/react-bootstrap/assets/1638860/0f659e3b-cd0d-4f7c-a6c5-18b6cb5da068">

**Comment by dobrienjr** [comment]
+1 - aria-label is not put on the element that has role=progressbar, instead it is put on a wrapping component.  Axe complains as well.
![image](https://github.com/user-attachments/assets/78a1b04f-0239-4b85-9f04-a4575996eeb2)
